src/CppUTestExt/MockSupport_c.cpp

```diff
@@ -311,7 +313,7 @@ MockExpectedCall_c* withConstPointerParameters_c(const char* name, const void* v
 
 MockExpectedCall_c* withFunctionPointerParameters_c(const char* name, void (*value)())
 {
-    expectedCall = &expectedCall->withParameter(name, value);
+    expectedCall = &expectedCall->withParameter(name, (Cpp_Func*)value);
```

Comment:
Again, it feels wrong to change the code here because of a quirk in just one compiler. Sometimes I feel like just dropping support for the Cl2000. Besides, these problems have been around for half a year and nobody has noticed.

I do wonder though about the whole idea of a function pointer parameter. Sure, I don't have to cast a void, void function anymore. But any other type of function with parameters, return value etc. I'd still have to force-cast to a basically incompatible signature, right?

So I was wondering, what is the advantage that this operation has over just casting to a generic pointer in the first place?